nexus/src/app/instance.rs

```diff
@@ -818,35 +869,70 @@ impl super::Nexus {
             Ok(Some(state)) => {
                 info!(
                     opctx.log,
-                    "instance terminated with extreme prejudice";
+                    "instance's {vmm_role} VMM terminated with extreme \
+                     prejudice";
                     "instance_id" => %authz_instance.id(),
                     "vmm_id" => %propolis_id,
                     "sled_id" => %sled_id,
                 );
 
-                self.notify_vmm_updated(opctx, propolis_id, &state).await?;
+                // We would like the caller to see the instance they are
+                // attempting to terminate go to "Stopped", so run the
+                // instance-update saga synchronously if possible. This is
+                // particularly important in the case where a migrating instance
+                // is force-terminated, as an instance with a migration ID will
+                // remain "Migrating" (not "Stopping") until its migration ID is
+                // unset, and it seems a bit sad to return a "Migrating"
+                // instance to a caller who tries to force-kill it.
+                //
+                // TODO(eliza): in the case where we are terminating both an
+                // active VMM and migration target, we will unregister them in
+                // sequence, running separate update sagas for both VMMs being
+                // terminated. It would be a bit more efficient to terminate
+                // both VMMs, update CRDB, and then run a *single* update saga
+                // to process both VMMs being destroyed. However, this requires
+                // a bit of annoying refactoring to existing APIs, and I'm not
+                // sure if improving the `instance-force-terminate` endpoint's
+                // latency is particularly important...
```

Comment:
I could be missing something, but it seems like we could at least return the generated `saga` from this function and then have the caller actually *execute* only one of the prepared sagas. (It'd be even better to call `update_saga_needed` just once, after applying both VMM changes, but I agree that that would be a bigger refactoring.)

Reply:
Done in https://github.com/oxidecomputer/omicron/pull/6795/commits/d69a1ceb3d9bd342a547747b4b5cfb3c52d6f2e8 --- can't believe I didn't think of that.